tech.json

```diff
@@ -861,6 +861,45 @@
               ]
             }
           ]
+        },
+        {
+          "name": "canEnterRMode",
+          "requires": [ "XRayScope", "ReserveTank", "canUseEnemies" ],
+          "note": [
+            "Ability to enter R-mode.",
+            "Have some energy in reserves (set to Auto), take damage while going through a door, use X-Ray while entering the next room, and release X-Ray after reserves have finished filling."
+          ],
+          "extensionTechs": [
+            {
+              "name": "canEnterGMode",
+              "requires": [ "canEnterRMode" ],
+              "note": [
+                "Ability to enter G-mode using a basic setup.",
+                "Have 4 energy in reserves (set to Auto), take damage while going through a door, use X-Ray while entering the next room, and release X-Ray before reserves have finished filling (a 4-frame window)."
+              ],
+              "extensionTechs": [
+                {
+                  "name": "canEnterGModeImmobile",
+                  "requires": [ "canEnterGMode" ],
+                  "note": [
+                    "Ability to enter G-mode immobile with high energy.",
+                    "Have 5 or more energy in reserves (set to Auto), take damage while going through a door, use X-Ray while entering the next room, and release X-ray immediately before reserves finish filling (normally a 3-frame window).",
+                    "Samus will be unable to move until getting hit by an enemy, or until canceling G-mode by using X-Ray."
+                  ]
+                },
+                {
+                  "name": "canArtificialMorph",
+                  "requires": [ "canEnterGMode" ],
+                  "note": [
+                    "Ability to obtain artificial morph with G-mode.",
+                    "Turn around after reserves finish filling but before knockback frames expire (a 4-frame window when entering G-mode with 4 reserve energy).",
+                    "For G-mode immobile, instead turn around while taking knockback from the enemy that returns control to Samus (a 5-frame window)",
+                    "This results in Samus entering a morph ball state without needing to have collected or equipped Morph."
+                  ]
```

Comment:
There may need to be new helpers for tech that require morph. Going through all the tech that require morph, it looks like only `canResetFallSpeed` and `canCrystalFlash` may be doable with artificial morph.

There are some helpers that might be worth looking at as well, although I imagine the gmode strats simply wont use these?: `h_canBombThings`, `h_canUsePowerBombs`, `h_canUseMorphBombs`, `h_canUseSpringBall`.

Reply:
I think canIBJ is another example and one which even applies to this room. I added a FIXME for those cases.